src/ProjectSystem/Impl/Extensions/ProjectTreeExtensions.cs

```diff
@@ -49,5 +49,16 @@ public static class ProjectTreeExtensions {
             }
             return Enumerable.Empty<string>();
         }
+
+        public static string GetSelectedFolderPath(this IImmutableSet<IProjectTree> nodes, UnconfiguredProject unconfiguredProject) {
+            if (nodes.Count == 1) {
+                var n = nodes.First();
+                if (n.Root == n) {
```

Comment:
Project node has Root set to itself (in which case I need to use different method for obtaining node path as project node in CPS return empty string from FilePath). Any better way to tell project node from the file node in CPS?